src/main/java/com/example/onlinebookstore/liquibase/UpdateUsersPassword.java

```diff
@@ -0,0 +1,116 @@
+package com.example.onlinebookstore.liquibase;
+
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import liquibase.change.custom.CustomTaskChange;
+import liquibase.database.Database;
+import liquibase.exception.CustomChangeException;
+import liquibase.exception.SetupException;
+import liquibase.exception.ValidationErrors;
+import liquibase.resource.ResourceAccessor;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+/**
+ * Class UpdateUsersPassword is used by Liquibase script to convert the previous simple user
+ * password to a new hashed password.
+ *
+ * There is no Spring context during the execution of execute method by liquibase script. This is
+ * why JDBC is uses to work with database.
+ *
+ * @author Maksym Bakunov
+ * @version 1.0
+ * @since 2024-10-05
+ */
+public class UpdateUsersPassword implements CustomTaskChange {
```

Comment:
Do you really need this class? 
when you register user you hash password and save it to DB. Looks like this logic is redundant. 
Please, text your answer and opinion in chat

Reply:
The conditions for this task contain the following item:

Add Liquibase changesets to insert roles into DB and assign role for existing users. Be sure you have at least one user in DB with role admin

This means that at the time of making new changes to the DB structure, the users table already contains user records, and therefore, at the time of making them, the password was not saved as a hash value. As for me, in addition to updating the table structures at this stage, it is necessary to bring the storage format of existing passwords into line. I decided to do this through liquibase. As for me, this option is more correct than manually converting passwords to their corresponding values ​​and then hardcoding them in scripts, since with a large number of existing users this will be tedious.

Reply:
we already discussed it in chat. So let's remove this migration ;)